### PR TITLE
search: Really show some error messages

### PR DESCRIPTION
**Pull request by marusak**
`result.reason` is an object and nothing was rendered.

### CODE REVIEW

File: src/ImageRunModal.jsx [hunk 1]
```diff
@@ -423,7 +423,7 @@ export class ImageRunModal extends React.Component {
                             } else {
                                 dialogError = _("Failed to search for new images");
                                 // TODO: add registry context, podman does not include it in the reply.
-                                dialogErrorDetail = cockpit.format(_("Failed to search for images: $0"), result.reason);
+                                dialogErrorDetail = result.reason ? cockpit.format(_("Failed to search for images: $0"), result.reason.message) : _("Failed to search for images.");
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:426]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1172-20230106-100359-1eb6bb49-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageSearchModal.jsx [hunk 2]
```diff
@@ -110,7 +110,7 @@ export class ImageSearchModal extends React.Component {
                                 results = results.concat(JSON.parse(result.value));
                             } else {
                                 dialogError = _("Failed to search for new images");
-                                dialogErrorDetail = cockpit.format(_("Failed to search for images: $0"), result.reason);
+                                dialogErrorDetail = result.reason ? cockpit.format(_("Failed to search for images: $0"), result.reason.message) : _("Failed to search for images.");
```
Comment (cockpituous) [inline src/ImageSearchModal.jsx:113]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1172-20230106-100359-1eb6bb49-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 1 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:426]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1172-20230106-105552-1eb6bb49-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageSearchModal.jsx (hunk 2 above)
Comment (cockpituous) [inline src/ImageSearchModal.jsx:113]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1172-20230106-105552-1eb6bb49-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]
Looks good, but now packit fails with these errors:

```
  File "/var/ARTIFACTS/work-allq0dyrr54/plans/all/discover/default-0/tests/test/common/testlib.py", line 1593, in tearDown
    self.check_journal_messages()
  File "/var/ARTIFACTS/work-allq0dyrr54/plans/all/discover/default-0/tests/test/common/testlib.py", line 1890, in check_journal_messages
    raise Error(UNEXPECTED_MESSAGE + "journal messages:\n" + first)
testlib.Error: FAIL: Test completed, but found unexpected journal messages:
(src/bridge/cockpithttpstream.c:942):cockpit_http_stream_close: code should not be reached
```

**Comment by jelly** [comment]
Looks go

> Looks good, but now packit fails with these errors:
> 
> ```
>   File "/var/ARTIFACTS/work-allq0dyrr54/plans/all/discover/default-0/tests/test/common/testlib.py", line 1593, in tearDown
>     self.check_journal_messages()
>   File "/var/ARTIFACTS/work-allq0dyrr54/plans/all/discover/default-0/tests/test/common/testlib.py", line 1890, in check_journal_messages
>     raise Error(UNEXPECTED_MESSAGE + "journal messages:\n" + first)
> testlib.Error: FAIL: Test completed, but found unexpected journal messages:
> (src/bridge/cockpithttpstream.c:942):cockpit_http_stream_close: code should not be reached
> ```

In other PR's it was just fedora-36, so this issue is really fun

**Comment by marusak** [comment]
> In other PR's it was just fedora-36, so this issue is really fun

yeah, it is some fun race somewhere :/

**Pull request merged**